### ISSUE TITLE
Typechecker should reject meth_caller for non-nullary methods

### ISSUE THREAD

**Issue by oyamauchi** [facebook/hhvm#4907]
If you create a meth_caller for a method that has nonzero arguments, there's no way to call it. The typechecker should probably throw an error if you try to do that.

```hack
class C {
  function f(int $x): void { }
}

function main(): void {
  $caller = meth_caller(C::class, 'f');  // <-- Should be an error
  $caller(/* ??? */);
}
```

**Comment by jwatzman** [comment]
Pretty sure this is fixed by 162a7893e163d50ddb791fcd0b258c081049fa07 and 89c4dedd5109b8a949e38a1a7e5d687a8ee0e8fa right?

**Comment by oyamauchi** [comment]
Turns out yes it is.
